src/main/java/shoai/Parser.java

```diff
@@ -0,0 +1,197 @@
+package shoai;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import shoai.TaskList;
+import shoai.Ui;
+import shoai.Storage;
+import shoai.Task;
+import shoai.Todo;
+import shoai.Deadline;
+import shoai.Event;
+import shoai.ShoAIException;
+import java.util.ArrayList;
+
+/**
+ * Parses user commands and executes the corresponding actions on the task list.
+ */
+public class Parser {
+
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    /**
+     * Parses a command and executes the appropriate action.
+     *
+     * @param fullCommand The full command string from the user.
+     * @param tasks The TaskList to operate on.
+     * @param ui The Ui instance for user interaction.
+     * @param storage The Storage instance for saving/loading tasks.
+     * @return true if the command is "bye" and the application should exit, false otherwise.
+     * @throws ShoAIException If there is an error processing the command.
+     */
+    public boolean parse(String fullCommand, TaskList tasks, Ui ui, Storage storage) throws ShoAIException {
```

Comment:
Perhaps a name more descriptive of the function contents should be used, given that this function does more than just parsing the argument. Maybe something like `processCommand` would be better?